OrbitService/OrbitAsioServer.cpp

```diff
@@ -24,7 +25,17 @@ void OrbitAsioServer::Run(std::atomic<bool>* exit_requested) {
   std::thread process_list_thread{
       [this, exit_requested] { ProcessListThread(exit_requested); }};
 
+  // make stdin non blocking
+  fcntl(STDIN_FILENO, F_SETFL, O_NONBLOCK);
+  // buffer for reading from stdin.
+  char buffer[10] = "";
+
   while (!(*exit_requested)) {
+    // read from stdin to buffer. This detect the potentially sent EOF
+    fgets(buffer, sizeof(buffer), stdin);
```

Comment:
while fgets(..) != nullptr) continue;

feof won't happen until you read everything from stdin.

Reply:
done